dbt/adapters/spark/connections.py

```diff
@@ -348,11 +349,21 @@ def open(cls, connection):
                     cls.validate_creds(creds,
                                        ['host', 'port', 'user', 'schema'])
 
-                    conn = hive.connect(host=creds.host,
-                                        port=creds.port,
-                                        username=creds.user,
-                                        auth=creds.auth,
-                                        kerberos_service_name=creds.kerberos_service_name)  # noqa
+                    if creds.use_ssl:
+                        import puretransport
```

Comment:
Let's move this import to the top. Since it will be an extra requirement, you'll want to structure it like:
```python
try:
    import puretransport
except ImportError:
    puretransport = None
```

Then down here, we can check for it:
```python
if creds.use_ssl:
    if not puretransport: # raise an error message
```